src/lib/progress.c

```diff
@@ -127,6 +131,40 @@ void progress_report(double count, double max)
 	}
 }
 
+int progress_spinner_callback()
+{
+	if (log_get_level() == LOG_DEBUG || !isatty(fileno(stdout))) {
+		return 0;
+	}
+
+	static int index;
+
+	if (begin == 0) {
```

Comment:
if you initialize begin here you will start printing the spinner after one second. you can just remove this if because time(NULL) will be always > 0 and begin will be initialized later

Reply:
Well, it will be 0 on 19 January 2038, but I think it's fine. hahaha